src/xrpld/rpc/detail/RPCHelpers.cpp

```diff
@@ -934,31 +940,32 @@ chooseLedgerEntryType(Json::Value const& params)
     std::pair<RPC::Status, LedgerEntryType> result{RPC::Status::OK, ltANY};
     if (params.isMember(jss::type))
     {
-        static constexpr std::array<std::pair<char const*, LedgerEntryType>, 22>
```

Comment:
Why was this type changed?

Reply:
It is not effective here. I suspect it was written before unordered_map appeared.

Reply:
`unordered_map` does not have a `constexpr` ctor. That's a significant change.

Reply:
The declaration was changed from `const` to `constexpr` only [about four years ago](https://github.com/XRPLF/rippled/commit/7724cca384487bf4cdc17fa34ff8d5372a54b110#diff-f268c49578e117687aef31d0b244a2bb193fee47fe364f6c822c587da7d76aa3L803), which is well after `unordered_map` appeared.

Reply:
@ximinez I agree with what you said, but I don't understand - is it objection to the change?

Reply:
Yeah. I think we're better off with a `constexpr` array than a non-`constexpr` `unordered_map`.

Reply:
reverted